core/os/mutex.h

```diff
@@ -117,4 +117,22 @@ using BinaryMutex = MutexImpl<FakeMutex>; // Non-recursive, handle with care
 
 #endif // !NO_THREADS
 
+template <class T>
+class LockGuard {
```

Comment:
Why not just use `MutexLock`?

Reply:
Ah I'll change this! :grinning: 

The lock guard was copied from the PR on PoolVector I was doing with @jordo, and we must have missed that there was a lock guard already. :+1: